src/test/java/de/cronn/validation_files_diff/impl/GoToValidationFileTest.java

```diff
@@ -0,0 +1,307 @@
+import de.cronn.validation_files_diff.JoiningStrategy;
+import de.cronn.validation_files_diff.action.GoToValidationFileAction;
+import de.cronn.validation_files_diff.action.GoToValidationFileHandler;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static com.intellij.codeInsight.navigation.GotoTargetHandler.GotoData;
+import static de.cronn.assertions.validationfile.TestData.TEST_OUTPUT_DATA_DIR;
+import static de.cronn.assertions.validationfile.TestData.TEST_VALIDATION_DATA_DIR;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+class GoToValidationFileTest extends AbstractValidationFilePluginTest implements JUnit5ValidationFileAssertions {
+	private static final String TEST_FILE_PATH = "src/test/Test.java";
+
+	private Path projectDir;
+
+	private GoToValidationFileAction action;
+	private GoToValidationFileHandler handler;
+	private GotoDataCaptor gotoDataCaptor;
+
+	@Override
+	@BeforeEach
+	protected void setUp() throws Exception {
+		super.setUp();
+
+		projectDir = createSampleProject();
+
+		action = spy(GoToValidationFileAction.class);
+		handler = spy(GoToValidationFileHandler.class);
+		when(action.getHandler()).thenReturn(handler);
+		gotoDataCaptor = new GotoDataCaptor();
+		doAnswer(gotoDataCaptor).when(handler).getSourceAndTargetElements(any(), any());
+	}
+
+	@Override
+	@AfterEach
+	protected void tearDown() throws Exception {
+		reset(action);
+		super.tearDown();
+	}
+
+	private static class GotoDataCaptor implements Answer<GotoData> {
+		private PsiElement[] targets;
+
+		private PsiElement[] getTargets() {
+			return targets;
+		}
+
+		@Override
+		public GotoData answer(InvocationOnMock invocationOnMock) throws Throwable {
+			GotoData result = (GotoData) invocationOnMock.callRealMethod();
+			if (result == null) {
+				return null;
+			}
+			targets = result.targets;
+			if (result.targets.length > 1) {
+				// Remove targets in order not to open a chooser popup window which would have to be handled in tests
+				result.targets = new PsiElement[]{ };
+			}
+			return result;
+		}
+	}
+
+	@ParameterizedTest
+	@CsvSource({
+			// folders
+			"<!C!>, ~o/Class", // output folder
+			"<!C!>, ~v/Class", // validation folder
+
+			// class
+			"<!C!>, ~v/Class/", // as directory
+			"<!C!>, ~v/Class2", // as prefix
+			"<!C!>, ~v/Class2/", // prefixed class as directory
+			"<!C!>, ~v/Class.txt", // with extension
+			"<!C!>, ~v/Class_other.txt", // with continuation
```

Comment:
As discussed in person, our project has use cases that may not be covered right now:

Example are validation files
* `data/test/validation/SepaPrenotificationIntegrationTest/Membership/testDownloadSepaPrenotification_noSepaPaymentMethod.json`
* `data/test/validation/SepaPrenotificationIntegrationTest/Membership/testDownloadSepaPrenotification_noSepaPaymentMethod_logging.txt`

for a test method `SepaPrenotificationIntegrationTest.Membership#testDownloadSepaPrenotification_noSepaPaymentMethod`

The important part is the custom suffix in the valildation file name that extends the method name. It is used to support multiple validation files per test in a lightweight way. Such a suffix may also come from parameterized tests.

The jump to the validation file works fine, but when jumping back to the method, it reaches the enclosing class.

Reply:
Background why this wasn't working is that it is somewhat a "mixed" strategy in creating validation file names. The main part is done with directory strategy (with "/" delimiter), while the suffix part is done with underscore strategy (with "_" delimiter). The better idea would be to add the different strategies already in the validation assertions library (where at the moment only the underscore strategy is included), but this is out of the scope of this pull request.

It is changed now in such a way that also mixing both strategies is allowed. See ce47e5182dac7d5a73b93df82cbcca5d0ed8883d

Reply:
Great, it works well! :slightly_smiling_face: